src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureSnapshotReference/SnapshotReference.cs

```diff
@@ -0,0 +1,27 @@
+using Azure.Data.AppConfiguration;
+using System.Threading;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Models
+{
+    /// <summary>
+    /// Represents a reference to a configuration snapshot that needs to be resolved.
+    /// </summary>
+    internal class SnapshotReference
+    {
+        public string SnapshotName { get; set; }
+
+        public ConfigurationClient Client { get; set; }
+
+        public CancellationToken CancellationToken { get; set; }
```

Comment:
I am surprised to see CancellationToken as a property on the snapshot reference. Why is that?